rpcs3/Emu/Cell/lv2/sys_spu.cpp

```diff
@@ -172,10 +171,30 @@ error_code sys_spu_image_open(vm::ptr<sys_spu_image> img, vm::cptr<char> path)
 	}
 
 	img->load(elf_file);
-
 	return CELL_OK;
 }
 
+error_code sys_spu_image_open(vm::ptr<sys_spu_image> img, vm::cptr<char> path)
+{
+	sys_spu.warning("sys_spu_image_open(img=*0x%x, path=%s)", img, path);
+
+	return spu_image_open(img, path.get_ptr());
+}
+
+error_code sys_spu_image_open_by_fd(vm::ptr<sys_spu_image> img, u32 fd, u64 offset)
+{
+	sys_spu.warning("sys_spu_image_open_by_fd(img=*0x%x, fd=%d)", img, fd);
+
+	const auto file = idm::get<lv2_fs_object, lv2_file>(fd);
+
+	if (!file)
+	{
+		return CELL_EBADF;
+	}
+
+	return spu_image_open(img, fmt::format("%s_x%x", file->name.data(), offset));
```

Comment:
Given the fd error check here, why not check for ENOENT in syst_spu_image_open and simplify spu_image_open by passing a file descriptor already pointing to the data?